code/game/machinery/doors/airlock_electronics.dm

```diff
@@ -3,100 +3,102 @@
+	/// Is the door access using req_one_access (TRUE) or req_access (FALSE)
+	var/one_access = FALSE
+	/// An associative list containing all station accesses. Includes their name and access number.
+	var/static/list/door_accesses_list = list()
+	/// Maximum brain damage a mob can have until it can't use the electronics
+	var/const/max_brain_damage = 60
+	/// Which direction has unrestricted access (e.g. medbay doors from the inside)
 	var/unres_direction = null
 
+/obj/item/airlock_electronics/Initialize(mapload)
+	. = ..()
+	if(!length(door_accesses_list))
+		for(var/access in get_all_accesses())
+			door_accesses_list += list(list(
+				"name" = get_access_desc(access),
+				"id" = access))
+
 /obj/item/airlock_electronics/attack_self(mob/user)
 	if(!ishuman(user) && !isrobot(user))
 		return ..()
 
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
 		if(H.getBrainLoss() >= max_brain_damage)
-			to_chat(user, "<span class='warning'>You forget how to use \the [src].</span>")
+			to_chat(user, "<span class='warning'>You forget how to use [src].</span>")
 			return
-
-	var/t1 = text("<B>Access control</B><br>\n")
-	t1 += "<hr>"
-	t1 += "<B> Unrestricted Access Settings</B><br>"
-
-	var/list/Directions = list("North","South",,"East",,,,"West")
-	for(var/direction in GLOB.cardinal)
-		if (unres_direction && unres_direction == direction)
-			t1 += "<a style='color: red' href='?src=[UID()];unres_direction=[direction]'>[Directions[direction]]</a><br>"
-		else
-			t1 += "<a href='?src=[UID()];unres_direction=[direction]'>[Directions[direction]]</a><br>"
-
-	t1 += "<hr>"
-	t1 += "Access requirement is set to "
-	t1 += one_access ? "<a style='color: green' href='?src=[UID()];one_access=1'>ONE</a><hr>" : "<a style='color: red' href='?src=[UID()];one_access=1'>ALL</a><hr>"
-
-	t1 += conf_access == null ? "<font color=red>All</font><br>" : "<a href='?src=[UID()];access=all'>All</a><br>"
-
-	var/list/accesses = get_all_accesses()
-	for(var/acc in accesses)
-		var/aname = get_access_desc(acc)
-
-		if(!conf_access || !conf_access.len || !(acc in conf_access))
-			t1 += "<a href='?src=[UID()];access=[acc]'>[aname]</a><br>"
-		else if(one_access)
-			t1 += "<a style='color: green' href='?src=[UID()];access=[acc]'>[aname]</a><br>"
-		else
-			t1 += "<a style='color: red' href='?src=[UID()];access=[acc]'>[aname]</a><br>"
-
-	t1 += "<p><a href='?src=[UID()];close=1'>Close</a></p>\n"
-
-	var/datum/browser/popup = new(user, "airlock_electronics", name, 400, 400)
-	popup.set_content(t1)
-	popup.open(0)
-	onclose(user, "airlock")
-
-/obj/item/airlock_electronics/Topic(href, href_list)
-	..()
-
-	if(usr.incapacitated() || (!ishuman(usr) && !isrobot(usr)))
-		return 1
-
-	if(href_list["close"])
-		usr << browse(null, "window=airlock_electronics")
+	tgui_interact(user)
+
+
+// tgui\packages\tgui\interfaces\AirlockElectronics.js
+/obj/item/airlock_electronics/tgui_interact(mob/user, ui_key, datum/tgui/ui, force_open, datum/tgui/master_ui, datum/tgui_state/state)
```

Comment:
This is missing all of the proper parameters and their definitions (EG: No state specified). Please look at another implemented TGUI DM side for a good example.